src/components/modals/TransferModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { EdgeAccount } from 'edge-core-js'
+import React from 'react'
+import { View } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import { cacheStyles } from 'react-native-patina'
+import AntDesignIcon from 'react-native-vector-icons/AntDesign'
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
+import { sprintf } from 'sprintf-js'
+
+import { selectWalletToken } from '../../actions/WalletActions'
+import { useHandler } from '../../hooks/useHandler'
+import { lstrings } from '../../locales/strings'
+import { config } from '../../theme/appConfig'
+import { useDispatch } from '../../types/reactRedux'
+import { getTokenId } from '../../util/CurrencyInfoHelpers'
+import { Airship, showError } from '../services/AirshipInstance'
+import { Theme, useTheme } from '../services/ThemeContext'
+import { ModalTitle } from '../themed/ModalParts'
+import { SelectableRow } from '../themed/SelectableRow'
+import { ThemedModal } from '../themed/ThemedModal'
+import { WalletListModal, WalletListResult } from './WalletListModal'
+
+interface Props {
+  bridge: AirshipBridge<void>
+  account: EdgeAccount
+  depositOrSend: 'deposit' | 'send'
+  navigation: any
+}
+
+interface TransferOption {
+  icon: React.ReactNode
+  title: string
+  onPress: () => void
+}
+
+/**
+ * Renders a modal with descritive options for transferring crypto in/out of
+ * Edge, i.e.: buying/selling/sending/receiving crypto
+ */
+export const TransferModal = ({ account, bridge, depositOrSend, navigation }: Props) => {
+  const theme = useTheme()
+  const style = styles(theme)
+  const dispatch = useDispatch()
+
+  const handleSell = useHandler(() => {
+    navigation.navigate('sellTab', { screen: 'pluginListSell' })
+    Airship.clear()
+  })
+  const handleBuy = useHandler(() => {
+    navigation.navigate('buyTab', { screen: 'pluginListBuy' })
+    Airship.clear()
+  })
+  const handleSend = useHandler(() => {
+    Airship.show<WalletListResult>(bridge => <WalletListModal bridge={bridge} headerTitle={lstrings.select_wallet} navigation={navigation} />)
+      .then(async ({ walletId, currencyCode }: WalletListResult) => {
```

Comment:
optional: use async/await syntax for readability

Reply:
also use

`.then(async ({ walletId, tokenId }:`

 instead of currencyCode. Then you don't need to call getTokenId